cpp/arcticdb/version/version_core.cpp

```diff
@@ -292,7 +292,11 @@ VersionedItem update_impl(
     util::check_rte(!index_segment_reader.is_pickled(), "Cannot update pickled data");
     auto index_desc = check_index_match(frame.index, index_segment_reader.tsd().stream_descriptor().index());
     util::check(index_desc.kind() == IndexDescriptor::TIMESTAMP, "Update not supported for non-timeseries indexes");
-    sorting::check<ErrorCode::E_UNSORTED_DATA>(frame.desc.get_sorted() == arcticdb::proto::descriptors::SortedValue::ASCENDING && std::holds_alternative<stream::TimeseriesIndex>(frame.index) && index_segment_reader.mutable_tsd().stream_descriptor().sorted() == arcticdb::proto::descriptors::SortedValue::ASCENDING,
+    sorting::check<ErrorCode::E_UNSORTED_DATA>(
+        (frame.desc.get_sorted() == arcticdb::proto::descriptors::SortedValue::ASCENDING || frame.desc.get_sorted() == arcticdb::proto::descriptors::SortedValue::UNKNOWN) && 
+        std::holds_alternative<stream::TimeseriesIndex>(frame.index) && 
+        (index_segment_reader.mutable_tsd().stream_descriptor().sorted() == arcticdb::proto::descriptors::SortedValue::ASCENDING || 
+            index_segment_reader.mutable_tsd().stream_descriptor().sorted() == arcticdb::proto::descriptors::SortedValue::UNKNOWN),
```

Comment:
I thought we agreed to switch to a state-machine-based approach?

If we're not there yet, the signal-to-noise ratio of these lines is still too low and the business logic is obscured.

Consider:
1. Extract the core decision logic to a free function with simple argument names like `existing_frame_sorted`, `input_frame_sorted`...
2. You can then use `if` statements to better specify the layers of decisions involved instead of having to cramp them all in one boolean expression.
3. `using SortedValue = arcticdb::proto::descriptors::SortedValue` in that function so to avoid the noisy prefixes
4. Do the complex object access (`index_segment_reader.mutable_tsd().stream_descriptor().sorted()`) on the calling side of that function

Reply:
even though, such function already exists, despite now it is only used by the append. Since initially, we wanted to force the update only work with sorted values, it is why in that case the state machine of the update is reduced to "resulting data always sorted", making that function unnecessary to be used there. Now, we will be able to use that. However, it lives in the write logic, so it has to be moved first to a free function. As Alex said, we decided to postpone it, in order to merge this PR ASAP.

Reply:
I think sacrificing code quality to meet arbitrary deadlines is bad. Doing so is creating technical debt which will ultimately add work for ourselves and may reduce our ability to react quickly in a real crisis.

Reply:
Yeah we didn't agree to do the absolute bare minimum here, we simply agreed to postpone the full refactor to the state machine. Cleaning this up with a function is a totally reasonable thing to do.

Reply:
yes, sorry. I was only referring to the state machine for the update (regarding only your question). The extraction of the code into a function is something that already was done yesterday. I agree it leaves the change cleaner.